tests/test_pyrflx.py

```diff
@@ -586,16 +620,484 @@ def test_ethernet_parse_binary(self) -> None:
+        )
+
+        self.ipv4.set("Version", 4)
+        self.ipv4.set("IHL", 5)
+        self.ipv4.set("DSCP", 0)
+        self.ipv4.set("ECN", 0)
+        self.ipv4.set("Total_Length", 44)
+        self.ipv4.set("Identification", 1)
+        self.ipv4.set("Flag_R", "False")
+        self.ipv4.set("Flag_DF", "False")
+        self.ipv4.set("Flag_MF", "False")
+        self.ipv4.set("Fragment_Offset", 0)
+        self.ipv4.set("TTL", 64)
+        self.ipv4.set("Protocol", "PROTOCOL_UDP")
+        self.ipv4.set("Header_Checksum", int("7CBE", 16))
+        self.ipv4.set("Source", int("7f000001", 16))
+        self.ipv4.set("Destination", int("7f000001", 16))
+        self.ipv4.set("Payload", data)
+
+        self.assertTrue(self.ipv4.valid_message)
+
+    def test_generating_ipv4_option(self) -> None:
+
+        self.ipv4_option.set("Copied", "False")
+        self.ipv4_option.set("Option_Class", "Debugging_And_Measurement")
+        self.ipv4_option.set("Option_Number", 4)
+        self.ipv4_option.set("Option_Length", 3)
+        self.ipv4_option.set("Option_Data", b"\x2a")
+
+        self.assertTrue(self.ipv4_option.valid_message)
+
+    def test_parsing_tlv_data(self) -> None:
+        test_bytes = b"\x40\x04\x00\x00\x00\x00"
+        self.tlv.assign(test_bytes)
+        self.assertTrue(self.tlv.valid_message)
+        self.assertEqual(test_bytes, self.tlv.bytestring)
+
+    def test_parsing_tlv_data_zero(self) -> None:
+        test_bytes = b"\x40\x00"
+        self.tlv_checksum.assign(test_bytes)
+        self.assertEqual(self.tlv_checksum.get("Tag"), "Msg_Data")
+        self.assertEqual(self.tlv_checksum.get("Length"), 0)
+        self.assertFalse(self.tlv_checksum.valid_message)
+
+    def test_parsing_tlv_error(self) -> None:
+
+        test_bytes = b"\xc0"
+        self.tlv_checksum.assign(test_bytes)
+        self.assertEqual(self.tlv_checksum.get("Tag"), "Msg_Error")
+        self.assertTrue(self.tlv_checksum.valid_message)
+
+    def test_parsing_invalid_tlv_invalid_tag(self) -> None:
+        test_bytes = b"\x00\x00"
+        with self.assertRaisesRegex(KeyError, "Number 0 is not a valid enum value"):
+            self.tlv_checksum.assign(test_bytes)
+
+    def test_generating_tlv_data(self) -> None:
+        expected = b"\x40\x04\x00\x00\x00\x00"
+
+        self.tlv.set("Tag", "Msg_Data")
+        self.tlv.set("Length", 4)
+        self.tlv.set("Value", b"\x00\x00\x00\x00")
+
+        self.assertTrue(self.tlv.valid_message)
+        self.assertEqual(self.tlv.bytestring, expected)
+
+    # def test_generating_tlv_data_zero(self) -> None:
+    # Message cannot become valid because length is 0 and thus
+    # value cannot be set
+
+    # expected = b"\x40\x00"
+    # self.tlv_checksum.set("Tag", "Msg_Data")
+    # self.tlv_checksum.set("Length", 0)
+    # self.tlv.set("Value", b"\x00\x00\x00\x00")
+    # self.assertTrue(self.tlv_checksum.valid_message)
+    # self.assertEqual(self.tlv_checksum.binary, expected)
```

Comment:
There is no issue assigned to this test. Please either assign an issue or remove the test altogether.

Reply:
Fixed by 30c636d